### PR TITLE
Add Select

### PR DESCRIPTION
**Pull request by Joozty**

### CODE REVIEW

**Comment by Joozty** [comment]
Feel free to take it I am ok as it looks/works now.

**Comment by Joozty** [comment]
@viktor-yakubiv I am missing button for clear input and apart from that, it can be reviewed. ( I will add it in this or following PR)

File: src/modules/select/select.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        {options.length && document.activeElement === inputRef.current
+          ? `${options.length} suggestions found, to navigate use up and down arrows`
+          : ''}
+      </div>
+      <Form.Group tag="div">
+        {prependIcon && (
+          <Form.Addon place="prepend">
+            <Icon src={prependIcon} />
+          </Form.Addon>
+        )}
```
[the remark this answers is not in the document]
Comment (Joozty) [reply src/modules/select/select.jsx:95]: well I had it as you wrote it before but you asked me to keep default styling of input fields, hence that big spacing. I will revert it back...

File: src/modules/select/utils/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export const KEYS = {
+  ESC: 27,
+  ENTER: 13,
+  LEFT: 37,
+  UP: 38,
+  RIGHT: 39,
+  DOWN: 40,
```
[the remark this answers is not in the document]
Comment (Joozty) [reply src/modules/select/utils/index.js:7]: what do you mean?

File: src/modules/select/select.jsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            place="append"
+            className={classNames.use(styles.clearOutIcon)}
+            onClick={() => {
+              setInputValue('')
+              inputRef.current.focus()
+            }}
+          >
+            <Icon src="#window-close" />
+          </Form.Addon>
+        )}
```
[the remark this answers is not in the document]
Comment (Joozty) [reply src/modules/select/select.jsx:107]: What DOM changes do you mean? It doesn't rerender only once when the input changes from empty string to something else.

[the remark this answers is not in the document]
Comment (viktor-yakubiv) [reply src/modules/select/select.jsx:107]: I mean exactly that when you change the input icon is created if it did not exist. It does not happen every time but why should you make the bowser recalculate everything if you can avoid it with a line of CSS?

Changing DOM impacts layout recalculation. It's heavy. The goal of React (according to the interview questions 😄) is to reduce the amount of DOM calculations but why should not we keep this in mind when we use React?

File: src/modules/select/select.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (viktor-yakubiv) [reply src/modules/select/select.jsx:95]: I meant the border and the size of the input. The icon bit is different

File: src/modules/select/utils/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (viktor-yakubiv) [reply src/modules/select/utils/index.js:7]: Check the example at the end of [this article](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key).

File: src/modules/select/select.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Joozty) [reply src/modules/select/select.jsx:107]: Interesting point 🤔  I wouldn't call it heavy thought. Nonetheless will hide it instead.

**Pull request merged**